src/Dialect/ONNX/ONNXOps/Additional/LayoutTransform.cpp

```diff
@@ -42,10 +42,9 @@ LogicalResult ONNXLayoutTransformOp::inferShapes(
 // Verifier
 //===----------------------------------------------------------------------===//
 LogicalResult ONNXLayoutTransformOp::verify() {
-  if (hasShapeAndRank(getData()) && hasShapeAndRank(getOutput())) {
-    if (getShape(getData().getType()) != getShape(getOutput().getType()))
-      return emitOpError("Input and output tensors must have the same shape");
-  }
+  if (auto dataType = getData().getType().dyn_cast<RankedTensorType>())
+    if (auto outputType = getOutput().getType().dyn_cast<RankedTensorType>())
+      return success(getShape(dataType) == getShape(outputType));
```

Comment:
@hamptonm1 does it fix your issue? or maybe we need to have a fine-grained check for each dimension of the input and output.
For example : input is `tensor<1x2x3xf32>` and output is `tensor<1x?x3xf32>`. In this case we return success because we cannot verify the real value of the output's second dim (that is unknown).

Reply:
Let me test this now!

Reply:
@tungld Does this update look better?

Reply:
@hamptonm1 it looks good. Thanks!